### DISCUSSION TITLE
How to Create a Video or GIF for Algorithms Visualization?

### DISCUSSION THREAD

**Discussion by UzairNaeem3** [AtsushiSakai/PythonRobotics#1115] [Q&A] [answered]
Hello,

I'm currently working with the Dijkstra algorithm implementation from this repository, and I've noticed that it generates separate plots for each step. However, in the documentation, I came across an excellent video demonstrating the algorithm's execution, which presents a continuous visualization. I have shared the animation below also.

I'm pursuing a degree in Mechatronics and am very interested in learning how to create a video or GIF that visually represents the step-by-step process of Dijkstra's algorithm. Could anyone share the technique used for this?

Thanks in advance!

![PythonRobotics/figure_1.png at master · AtsushiSakai/PythonRobotics](https://github.com/AtsushiSakai/PythonRoboticsGifs/raw/master/PathPlanning/Dijkstra/animation.gif)

**Comment by AtsushiSakai** [comment, marked as answer]
You can use this script: https://github.com/AtsushiSakai/matplotrecorder

**Reply by UzairNaeem3** [reply]
Thank you so much!

**Reply by UzairNaeem3** [reply]
This is awesome! Just out of curiosity, I'm thinking the same thing can be achieved with OpenCV, which would be helpful for Windows-based users. I am planning to achieve the same result using OpenCV and would love to create a PR for the new script if you're open to it.
